x-pack/plugins/observability/public/pages/slo_edit/components/custom_common/index_selection.tsx

```diff
@@ -5,144 +5,124 @@
  * 2.0.
  */
 
-import { EuiComboBox, EuiComboBoxOptionOption, EuiFormRow } from '@elastic/eui';
+import { EuiFormRow } from '@elastic/eui';
 import { DataView } from '@kbn/data-views-plugin/public';
 import { i18n } from '@kbn/i18n';
-import { debounce } from 'lodash';
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { Controller, useFormContext } from 'react-hook-form';
+import { DataViewPicker } from '@kbn/unified-search-plugin/public';
+import { useKibana } from '../../../../utils/kibana_react';
+import { ObservabilityPublicPluginsStart } from '../../../..';
 import { useFetchDataViews } from '../../../../hooks/use_fetch_data_views';
-import { useFetchIndices } from '../../../../hooks/use_fetch_indices';
 import { CreateSLOForm } from '../../types';
 
-interface Option {
-  label: string;
-  options: Array<{ value: string; label: string }>;
-}
-
 export function IndexSelection() {
-  const { control, getFieldState } = useFormContext<CreateSLOForm>();
-  const [searchValue, setSearchValue] = useState<string>('');
+  const { control, getFieldState, setValue, watch } = useFormContext<CreateSLOForm>();
+  const { dataViews: dataViewsService } = useKibana().services;
 
-  const { isLoading: isIndicesLoading, data: indices = [] } = useFetchIndices({
-    search: searchValue,
-  });
-  const { isLoading: isDataViewsLoading, data: dataViews = [] } = useFetchDataViews({
-    name: searchValue,
-  });
+  const { isLoading: isDataViewsLoading, data: dataViews = [] } = useFetchDataViews();
 
-  const options: Option[] = [];
-  if (!isDataViewsLoading && dataViews.length > 0) {
-    options.push(createDataViewsOption(dataViews));
-  }
-  if (!isIndicesLoading && !!searchValue) {
-    options.push(createIndexPatternOption(searchValue, indices));
-  }
+  const { dataViewEditor } = useKibana<ObservabilityPublicPluginsStart>().services;
+
+  const [adHocDataViews, setAdHocDataViews] = useState<DataView[]>([]);
+
+  const currentIndexPattern = watch('indicator.params.index');
 
-  const onSearchChange = debounce((value: string) => setSearchValue(value), 300);
+  useEffect(() => {
+    const missingAdHocDataView = adHocDataViews.find(
+      (dataView) => dataView.getIndexPattern() === currentIndexPattern
+    );
 
-  const placeholder = i18n.translate('xpack.observability.slo.sloEdit.indexSelection.placeholder', {
-    defaultMessage: 'Select an index pattern',
-  });
+    if (!missingAdHocDataView && currentIndexPattern) {
+      async function loadMissingDataView() {
+        const dataView = await dataViewsService.create(
+          {
+            title: currentIndexPattern,
+            allowNoIndex: true,
+          },
+          true
+        );
+        if (dataView.getIndexPattern() === currentIndexPattern) {
+          setAdHocDataViews((prev) => [...prev, dataView]);
+        }
+      }
+      loadMissingDataView();
+    }
+  }, [adHocDataViews, currentIndexPattern, dataViewsService]);
```

Comment:
If you changed this `useEffect` this:
```typescript
  useEffect(() => {
    if (!isDataViewsLoading) {
      const missingAdHocDataView =
        dataViews.find((dataView) => dataView.title === currentIndexPattern) ||
        adHocDataViews.find((dataView) => dataView.getIndexPattern() === currentIndexPattern);

      if (!missingAdHocDataView && currentIndexPattern) {
        async function loadMissingDataView() {
          const dataView = await dataViewsService.create(
            {
              title: currentIndexPattern,
              allowNoIndex: true,
            },
            true
          );
          if (dataView) {
            setAdHocDataViews((prev) => [...prev, dataView]);
          }
        }
        loadMissingDataView();
      }
    }
  }, [adHocDataViews, currentIndexPattern, dataViewsService, isDataViewsLoading, dataViews]);
```

Then it wouldn't list the index pattern as an adhoc index pattern when a suitable DataView exists.